WaterLog_Backend/Controllers/ReadingsController.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.Extensions.Configuration;
+using WaterLog_Backend.Models;
+
+namespace WaterLog_Backend.Controllers
+{
+
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ReadingsController : ControllerBase
+    {
+        private readonly DatabaseContext _db;
+        readonly IConfiguration _config;
+        private IControllerService _service;
+
+        public ReadingsController(DatabaseContext context, IConfiguration config, IControllerService service)
+        {
+            _db = context;
+            _config = config;
+            _service = service;
+        }
+
+        // GET api/values
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<ReadingsEntry>>> Get()
+        {
+            return await _db.Readings.ToListAsync();
+        }
+
+        // GET api/values/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<ReadingsEntry>> Get(int id)
+        {
+            return await _db.Readings.FindAsync(id);
+        }
+
+        // POST api/values
+        [HttpPost]
+        public async Task Post([FromBody] ReadingsEntry value)
+        {
+            value.TimesStamp = DateTime.UtcNow;
+            await _db.Readings.AddAsync(value);
+            await _db.SaveChangesAsync();
+            Procedures procedure = new Procedures(_db, _config);
+            await procedure.triggerInsert(value);
+        }
+
+        // PUT api/values/5
+        [HttpPut("{id}")]
+        public async Task Put(int id, [FromBody] ReadingsEntry value)
+        {
+            var old = await _db.Readings.FindAsync(id);
+            _db.Entry(old).CurrentValues.SetValues(value);
```

Comment:
Quick question, are we going to have any error handling? I'm not asking for code change, maybe you can tell me what your plans are and with an exception of that I'll resolve this comment.

Reply:
@ash-rabbit We will put error handling on this. Will let you know when it has been done.

Reply:
Cool thanks